src/mappings/entities/mapSettlement.ts

```diff
@@ -53,8 +54,14 @@ export const createLeg = async (
   address: string,
   legIndex: number,
   { ticker, amount, from, to }: LegDetails
-): Promise<void> =>
-  Leg.create({
+): Promise<void> => {
+  // since an instruction leg can be created without a valid DID, we make sure DB has an entry for Identity to avoid foreign key constraint
```

Comment:
I think we should make the reference nullable instead of making an Identity that doesn't really exist.

DID's are quasi-random, so its very unlikely someone predicted their counter party. If this happens its likely to be bad user input.

I think we should keep the shortcoming of the Leg to the leg, and not let it weaken the Identity model. I think having two fields one a nullable foreign reference is better than inserting Identities that don't really exist.